src/extension.ts

```diff
@@ -4,148 +4,194 @@
+        synchronize: {
+            // Notify the server about file changes to 'smithy-build.json' or 'smithy.json' files contained in the workspace
+            fileEvents: workspace.createFileSystemWatcher('**/{smithy-build}.json')
+        }
+    };
+
+    // Create the language client and start the client.
+
+    client = new LanguageClient(
+        'smithyLsp',
+        'Smithy LSP',
+        createServer,
+        clientOptions
+    );
+    const smithyContentProvider = createSmithyContentProvider(client);
+    context.subscriptions.push(workspace.registerTextDocumentContentProvider('smithyjar', smithyContentProvider));
+
+    // Start the client. This will also launch the server
+    client.start();
 }
 
 export function deactivate(): Thenable<void> | undefined {
-	if (!client) {
-		return undefined;
-	}
-	return client.stop();
+    if (!client) {
+        return undefined;
+    }
+    return client.stop();
 }
 
 function createSmithyContentProvider(languageClient: LanguageClient): vscode.TextDocumentContentProvider {
-	return <vscode.TextDocumentContentProvider>{
-		provideTextDocumentContent: async (uri: vscode.Uri, token: CancellationToken): Promise<string> => {
-			return languageClient.sendRequest(ClassFileContentsRequest.type, { uri: uri.toString() }, token).then((v: string): string => {
-				return v || '';
-			});
-		}
-	}
+    return <vscode.TextDocumentContentProvider>{
+        provideTextDocumentContent: async (uri: vscode.Uri, token: CancellationToken): Promise<string> => {
+            return languageClient.sendRequest(ClassFileContentsRequest.type, { uri: uri.toString() }, token).then((v: string): string => {
+                return v || '';
+            });
+        }
+    }
+}
+
+function getCoursier(extensionPath: string): Promise<string> {
+    function binPath(filename: string) { 
+        return path.join(extensionPath, filename);
+    }
+    const urls = {
+        darwin: "https://github.com/coursier/coursier/releases/download/v2.0.6/cs-x86_64-apple-darwin",
+        linux: "https://github.com/coursier/coursier/releases/download/v2.0.6/cs-x86_64-pc-linux",
+        win32: "https://github.com/coursier/coursier/releases/download/v2.0.6/cs-x86_64-pc-win32.exe"
+    }
+    const targets = {
+        darwin: binPath("coursier"),
+        linux: binPath("coursier"),
+        win32: binPath("coursier.exe")
+    }
+    return downloadFile(urls[process.platform], targets[process.platform])
+}
+
+function downloadFile(url: string, targetFile: string): Promise<string> {
+    function get(url: string, resolve: (res: string) => void, reject: (err: Error) => void) {
+        https.get(url, response => {
+            if(response.statusCode === 301 || response.statusCode === 302) {
+                const newUrl = response.headers.location;
+                console.log(`Following redirection to ${newUrl}.`);
+                return get(newUrl, resolve, reject)

```

Comment:
**nitpick:** is there any way we could tell `https` to follow redirects automatically? Or use some lightweight library that would take care of details like that

Reply:
yeah, there are multiple libs, will do

Reply:
done, used follow-redirects